src/main/java/gregtech/api/render/MetaTileEntityTESR.java

```diff
@@ -14,22 +15,65 @@
-        if(te.getMetaTileEntity() instanceof IFastRenderMetaTileEntity) {
+        MetaTileEntity metaTileEntity = te.getMetaTileEntity();
+        if(metaTileEntity instanceof IFastRenderMetaTileEntity) {
             renderTileEntityFastPart(te, x, y, z, partialTicks, destroyStage);
         }
-        if(te instanceof IRenderMetaTileEntity) {
-            MetaTileEntity metaTileEntity = te.getMetaTileEntity();
-            if (metaTileEntity instanceof IRenderMetaTileEntity) {
-                ((IRenderMetaTileEntity) metaTileEntity).renderMetaTileEntityDynamic(x, y, z, partialTicks);
+        if(metaTileEntity instanceof IRenderMetaTileEntity) {
+            ((IRenderMetaTileEntity) metaTileEntity).renderMetaTileEntityDynamic(x, y, z, partialTicks);
+        }
+        if(metaTileEntity != null) {
+            List<Tuple<IFastRenderMetaTileEntity, EnumFacing>> coverFast = new ArrayList<>();
+            for (EnumFacing side: EnumFacing.VALUES){
+                CoverBehavior cover = metaTileEntity.getCoverAtSide(side);
+                if (cover instanceof IFastRenderMetaTileEntity) {
+                    coverFast.add(new Tuple<>((IFastRenderMetaTileEntity) cover, side));
+                } else if (cover instanceof IRenderMetaTileEntity){
+                    ((IRenderMetaTileEntity) cover).renderMetaTileEntityDynamic(x, y, z, partialTicks);
+                }
+            }
+            if (coverFast.size() > 0) {
+                Matrix4 translation = (new Matrix4()).translate(x, y, z);
+                Tessellator tessellator = Tessellator.getInstance();
+                BufferBuilder buffer = tessellator.getBuffer();
+                this.bindTexture(TextureMap.LOCATION_BLOCKS_TEXTURE);
+                RenderHelper.disableStandardItemLighting();
+                GlStateManager.blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
+                GlStateManager.enableBlend();
+
+                if (Minecraft.isAmbientOcclusionEnabled()) {
+                    GlStateManager.shadeModel(GL11.GL_SMOOTH);
+                }
+                else {
+                    GlStateManager.shadeModel(GL11.GL_FLAT);
+                }
+                buffer.begin(GL11.GL_QUADS, DefaultVertexFormats.BLOCK);
+
+                for (Tuple<IFastRenderMetaTileEntity, EnumFacing> tuple : coverFast) {
+                    CCRenderState renderState = CCRenderState.instance();
+                    renderState.reset();
+                    renderState.bind(buffer);
+                    renderState.setBrightness(te.getWorld(), te.getPos().offset(tuple.getSecond()));
+                    tuple.getFirst().renderMetaTileEntityFast(renderState, translation, partialTicks);
+                }
+
+                buffer.setTranslation(0, 0, 0);
+                tessellator.draw();
+                RenderHelper.enableStandardItemLighting();
```

Comment:
There should be a call to `GlStateManager.enableCull()` here as well, right before this line